infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/GeocodingDao.kt

```diff
@@ -80,67 +81,88 @@ class GeocodingDao(
     }
 
     fun getLayoutGeocodingContextCacheKey(
+        branch: LayoutBranch,
         trackNumberId: IntId<TrackLayoutTrackNumber>,
         moment: Instant,
     ): GeocodingContextCacheKey? {
         //language=SQL
         val sql = """
             with 
-              tn as (
-                select id, version, state, deleted 
+              tn_versions as (
```

Comment:
Ajanhetki-haut tehdään nyt aina 2 stepillä (saa ehdottaa jos joku keksii paremman tavan): ensimmäisessä haetaan tuorein versio kaikista tarkastelulle oleellisista riveista (design tai main) ja toisessa poimitaan niistä se oikea. Oleellista tässä on että:
- row versioista täytyy poimia myös deleted versiot, koska jos uusin (mukana oleva) on deleted, koko käsite on deleted ja sen aiemmatkin versiot pitää jättää huomiotta. Suodatus tuolla siksi vasta 2-vaiheessa
- design-käsite voi siirtyä main käsitteeksi mutta ei toisin päin -> näissä uusin versio on main ja aiemmat voi siis jättää huomiotta
- Jos haetaan designia, niin olemassaoleva design-rivi on se oikea jos sellainen löytyy. Muutoin aina main.